homeassistant/components/sensor/filesize.py

```diff
@@ -0,0 +1,105 @@
+"""
+Sensor for monitoring the size of a file.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.filesize/
+"""
+import datetime
+import logging
+import os
+
+import voluptuous as vol
+
+from homeassistant.helpers.entity import Entity
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+
+_LOGGER = logging.getLogger(__name__)
+
+
+CONF_FILE_PATHS = 'file_paths'
+ICON = 'mdi:file'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_FILE_PATHS):
+        vol.All(cv.ensure_list, [cv.isfile]),
+})
+
+
+def get_file_size(path):
+    """Return the size of the file in bytes."""
+    statinfo = os.stat(path)
+    file_size_bytes = statinfo.st_size
+    return file_size_bytes
+
+
+def get_last_updated(path):
+    """Return the time the file was last modified."""
+    statinfo = os.stat(path)
```

Comment:
You're calling `os.stat` twice per update because you made 2 functions. Why not just keep this all inside the `update` method?

Reply:
Good point, done